Polyfills/Console/Source/Console.cpp

```diff
@@ -34,16 +38,136 @@ namespace
+            {
+                const char currChar = firstArg.at(j);
+                if (!matching && currChar == '%')
+                {
+                    matching = true;
+                    if (j > 0)
+                    {
+                        std::string_view prefix(firstArg.data() + start, j - start);
+                        ss << prefix;
+                    }
+                    start = j;
+                }
+                else if (matching && (currChar == 's' || currChar == 'o' || currChar == 'O' || currChar == 'd' || currChar == 'i' || currChar == 'f'))
+                {
+                    matching = false;
+                    std::string_view prefix(firstArg.data() + start, j - start + 1);
+                    // This is a substitution argument, so if we have any remaining arguments to substitute, we should.
+                    // Try to use the next arg for subsitution
+                    if (currArgIndex < info.Length())
+                    {
+                        Napi::Value currArg = info[currArgIndex];
+                        // Check the type of the sub string
+                        if (EndsWith(prefix, 'f'))
+                        {
+                            double d = currArg.ToNumber().DoubleValue();
+                            if (std::isnan(d))
+                            {
+                                ss << "NaN";
+                            }
+                            else
+                            {
+                                // I had to copy the string view to a string here, because the data pointer of the string view
+                                // returns a view of the entire character sequence, which caused an issue on the following
+                                // step when formatting.
+                                std::string copiedSubString(prefix);
+                                std::string formatted = FormatValue(copiedSubString, d);
+                                ss << formatted;
+                            }
+                        }
+                        else if (EndsWith(prefix, 'd') || EndsWith(prefix, 'i'))
+                        {
+                            // For some reason, converting to int doesn't result in nans, so I check with double.
+                            double d = currArg.ToNumber().DoubleValue();
+                            if (std::isnan(d))
+                            {
+                                ss << "NaN";
+                            }
+                            else
+                            {
+                                int64_t n = currArg.ToNumber().Int64Value();
+                                // Some explanation as above to why copy to a string
+                                std::string copiedSubString(prefix);
+                                std::string formatted = FormatValue(copiedSubString, n);
+                                ss << formatted;
+                            }
```

Comment:
Some relevant info:
- JS has no concept of integers. All numbers are doubles. Node-API will coerce the JS double value to a C integer value in `Int64Value()`.
- NaN is a floating-point concept. Integers don't have NaN.

Probably what should happen here is call `DoubleValue()` first to check for NaN like you are doing and then cast the result to integer if the currChar is `i`. Right now, this code will always format as integer even if it's double.